x-pack/filebeat/input/awss3/states.go

```diff
@@ -78,6 +78,12 @@ func (s *states) MustSkip(state state, store *statestore.Store) bool {
 		return true
 	}
 
+	// the previous state is stored or has error: let's skip
```

Comment:
there's quite a few places where this combined check is now performed. what do you think about a smaller method in state.go to simplify the logic. something like: 

```
--- a/x-pack/filebeat/input/awss3/s3.go
+++ b/x-pack/filebeat/input/awss3/s3.go
@@ -281,8 +281,8 @@ func (p *s3Poller) Purge() {
 		latestStoredTimeByBucketAndListPrefix := make(map[string]time.Time, 0)
 
 		for _, state := range p.states.GetStatesByListingID(listingID) {
-			// it is not stored, keep
-			if !state.Stored && !state.Error {
+			// it is not processed, keep
+			if !state.IsProcessed() {
 				p.log.Debugw("state not stored or with error, skip purge", "state", state)
 				continue
 			}

--- a/x-pack/filebeat/input/awss3/state.go
+++ b/x-pack/filebeat/input/awss3/state.go
@@ -60,6 +60,11 @@ func (s *state) MarkAsError() {
 	s.Error = true
 }
 
+// IsProcessed checks if the state is either Stored or Error
+func (s *state) IsProcessed() bool {
+    return s.Stored || s.Error
+}
+
 // IsEqual checks if the two states point to the same s3 object.
 func (s *state) IsEqual(c *state) bool {
 	return s.Bucket == c.Bucket && s.Key == c.Key && s.Etag == c.Etag && s.LastModified.Equal(c.LastModified)

--- a/x-pack/filebeat/input/awss3/states.go
+++ b/x-pack/filebeat/input/awss3/states.go
@@ -78,15 +78,15 @@ func (s *states) MustSkip(state state, store *statestore.Store) bool {
 		return true
 	}
 
-	// the previous state is stored or has error: let's skip
-	if !previousState.IsEmpty() && (previousState.Stored || previousState.Error) {
+	// the previous state is already processed: let's skip
+	if !previousState.IsEmpty() && previousState.IsProcessed() {
 		s.log.Debugw("previous state is stored or has error", "state", state)
 		return true
 	}
 
 	// we have no previous state or the previous state
-	// is not stored: refresh the state
-	if previousState.IsEmpty() || (!previousState.Stored && !previousState.Error) {
+	// is not processed: refresh the state
+	if previousState.IsEmpty() || !previousState.IsProcessed() {
 		s.Update(state, "")
 	}
 
@@ -172,7 +172,7 @@ func (s *states) Update(newState state, listingID string) {
 		s.log.Debug("New state added for ", newState.ID)
 	}
 
-	if listingID == "" || (!newState.Stored && !newState.Error) {
+	if listingID == "" || !newState.IsProcessed() {
 		return
 	}
 ```

Reply:
done